### PR TITLE
Add Jest tests coverage to CodeClimate

### PR DESCRIPTION
**Pull request by valerian**
## Work performed

- Change Jest tests config to generate an lcov coverage report
- Add a CI action to publish lcov data to CodeClimate

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -159,6 +159,30 @@ jobs:
         run: yarn test
         env:
           POSTGRES_EXTERNAL_URL: postgresql://admin:admin@localhost:5433/labelflow?schema=public
+      - name: upload coverage artifact
```
Comment (bbenoist) [inline .github/workflows/ci.yml:162]:
```suggestion
      - name: Upload code-coverage output
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -159,6 +159,30 @@ jobs:
...
+        uses: actions/upload-artifact@v2
+        with:
+          name: jest code coverage
```
Comment (bbenoist) [inline .github/workflows/ci.yml:165]:
```suggestion
          name: lcov.info
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -159,6 +159,30 @@ jobs:
...
+    runs-on: ubuntu-20.04
+    needs: test-typescript
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+      - name: download code coverage artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: jest code coverage
+      - name: publish jest code coverage to codeclimate
```
Comment (bbenoist) [inline .github/workflows/ci.yml:179]: Publish code-coverage output to CodeClimate

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -159,6 +159,30 @@ jobs:
...
+        uses: paambaati/codeclimate-action@v3.0.0
+        env:
+          CC_TEST_REPORTER_ID: ${{secrets.CC_TEST_REPORTER_ID}}
+        with:
+          coverageLocations: |
+            lcov.info:lcov
```
Comment (bbenoist) [inline .github/workflows/ci.yml:185]: What is this for?

File: package.json [hunk 5]
```diff
@@ -12,8 +12,8 @@
     "lint:md": "markdownlint **/*.md",
     "lint:ts": "eslint '{cypress,typescript}/**/*.{ts,tsx}'",
     "lint:all": "yarn lint",
-    "test": "dotenv -e .env.test -- jest --runInBand",
-    "test:all": "dotenv -e .env.test -- jest --runInBand",
+    "test": "dotenv -e .env.test -- jest --coverage --runInBand",
+    "test:all": "dotenv -e .env.test -- jest --coverage --runInBand",
```
Comment (bbenoist) [inline package.json:16]:
```suggestion
    "test": "dotenv -e .env.test -- jest --runInBand",
    "test:all": "dotenv -e .env.test -- jest --runInBand",
```

**Review by bbenoist — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -159,6 +159,30 @@ jobs:
...
+    name: Publish tests code coverage to Code Climate
+    runs-on: ubuntu-20.04
+    needs: test-typescript
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v2
+      - name: download code coverage artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: jest code coverage
```
[the remark this answers is not in the document]
Comment (bbenoist) [reply .github/workflows/ci.yml:178]: I was messy during my review. Indeed it's a suggestion.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (bbenoist) [reply .github/workflows/ci.yml:179]: I was messy during my review. Indeed it's a suggestion.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (valerian) [reply .github/workflows/ci.yml:185]: for information the `upload-artifact` doesn't keep the full path of `coverage/lcov.info`, when I do `download-artifact` I get a `lcov.info` put at the root folder.

Comment (bbenoist) [reply .github/workflows/ci.yml:185]: Yes, that's normal since you're using an artifact instead of the cache 😉

**Review by bbenoist — APPROVED** [review]

**Pull request merged**